contracts/EVault.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@rari-capital/solmate/src/tokens/ERC20.sol";
+import "@rari-capital/solmate/src/utils/SafeTransferLib.sol";
+//important! Importing this library from solmate will cause the contract to fail compile
+import "./utils/ERC4626.sol";
+import "./utils/FixedPointMathLib.sol";
+
+
+// An ERC4626 compliant vault that interacts with a strategy address
+// BUFFER is the minimun amount of tokens that can be stored in the vault and should
+// be compared with bufBal to determine if the vault neeed topup
+// TOP_UP indicates if funds are needed from bot to facilitate a large withdrawal
+ contract EVault is ERC4626 {
+    using SafeTransferLib for ERC20;
+    //lets track balances for now
+    mapping(address => uint256) balances;
```

Comment:
I don't think you will need balances since the actual ERC4626 contract mints shares based on balances. Can scrap this.